x-pack/plugins/enterprise_search/server/routes/enterprise_search/config_data.ts

```diff
@@ -17,7 +17,12 @@ export function registerConfigDataRoute({ router, config, log }: RouteDependenci
     async (context, request, response) => {
       const data = await callEnterpriseSearchConfigAPI({ request, config, log });
 
-      if (!Object.keys(data).length) {
+      if ('responseStatus' in data) {
```

Comment:
I think we can merge this if status with the one on the bottom
```javascript
      if ('responseStatus' in data || !Object.keys(data).length) {
        return response.customError({
          statusCode: data.responseStatus ?? 502,
          body: 'Error fetching data from Enterprise Search',
        });
      } else if ...
```

Reply:
This gives me a linter error as well.
<img width="666" alt="image" src="https://user-images.githubusercontent.com/637013/166817962-fbb8746c-6e20-4a70-aea2-a6b3ed9aea38.png">

Reply:
After running the tests, it works anyway. 🤷 So I'll use your version, shorter is better. Linter is stupid.

Reply:
Nope. Kibana tests broke after I applied your suggestion.

https://buildkite.com/elastic/kibana-pull-request/builds/42647#cba59332-0984-47d0-8229-b578e362bb63

Reverting back to my old version.